### PR TITLE
V0.12 issue 295

### PR DESCRIPTION
**Pull request by dannyleesmith**
##### Corresponding Issue(s):

https://github.com/rackspace-infrastructure-automation/aws-terraform-internal/issues/295

##### Summary of change(s):

- Add identity output
- Use tags
- Managed CloudWatch Log Group

##### Reason for Change(s):

- QOL

##### Will the change trigger resource destruction or replacement? If yes, please provide justification:

No

##### Does this update/change involve issues with other external modules? If so, please describe the scenario.

No

##### If input variables or output variables have changed or has been added, have you updated the README?

Updated

##### Do examples need to be updated based on changes?

Done

##### Note to the PR requester about Closing PR's

Please message the person that opened the issue when auto closing it on slack, as well as any other stake holders of deep interest. Only close the issue if you believe that the issue is fully resolved with this PR.

#### This PR may auto close the issue associated with it. If you feel the issue is not resolved please reopen the issue.

### CODE REVIEW

File: modules/cluster/examples/example.tf [hunk 1]
```diff
@@ -18,11 +18,11 @@ locals {
 module "vpc" {
   source = "git@github.com:rackspace-infrastructure-automation/aws-terraform-vpc_basenetwork?ref=v0.12.0"
 
-  name = "Test1VPC"
-
   custom_tags = {
     "kubernetes.io/cluster/${local.eks_cluster_name}" = "shared"
   }
+
+  name = "Test1VPC"
```
Comment (johnctitus) [inline modules/cluster/examples/example.tf]: Nitpicky, but we should revert this change as multiline properties should come after all single line properties

File: modules/cluster/main.tf [hunk 2]
```diff
@@ -85,6 +96,15 @@ resource "aws_iam_role_policy_attachment" "attach_worker_policy" {
   )
 }
 
+resource "aws_cloudwatch_log_group" "eks" {
+  count = var.manage_log_group ? 1 : 0
+
+  name              = "/aws/eks/${var.name}/cluster"
+  retention_in_days = var.log_group_retention
+
+  tags = local.merged_tags
```
Comment (johnctitus) [inline modules/cluster/main.tf]: Optional, but I would not separate these lines
```suggestion
  retention_in_days = var.log_group_retention
  tags              = local.merged_tags
```

File: modules/cluster/main.tf [hunk 3]
```diff
@@ -95,6 +115,12 @@ resource "aws_eks_cluster" "cluster" {
     security_group_ids = var.security_groups
     subnet_ids         = var.subnets
   }
+
+  tags = local.merged_tags
```
Comment (johnctitus) [inline modules/cluster/main.tf]: Nitpicky, but this should go between lines 111 and 112

File: modules/cluster/variables.tf [hunk 4]
```diff
@@ -29,12 +29,30 @@ variable "enabled_cluster_log_types" {
   ]
 }
 
+variable "environment" {
+  description = "Application environment for which this network is being created. Preferred value are Development, Integration, PreProduction, Production, QA, Staging, or Test"
+  type        = "string"
```
Comment (johnctitus) [inline modules/cluster/variables.tf]: Minor change due to HCL2
```suggestion
  type        = string
```

File: modules/cluster/variables.tf [hunk 5]
```diff
@@ -29,12 +29,30 @@ variable "enabled_cluster_log_types" {
...
+
 variable "kubernetes_version" {
   description = "The desired Kubernetes version for your cluster. If you do not specify a value here, the latest version available in Amazon EKS is used."
   type        = string
   default     = ""
 }
 
+variable "log_group_retention" {
+  description = "Specifies the number of days you want to retain log events in the specified log group. Possible values are: 0 (Never Expire), 1, 3, 5, 7, 14, 30, 60, 90, 120, 150, 180, 365, 400, 545, 731, 1827, and 3653."
+  type        = "string"
```
Comment (johnctitus) [inline modules/cluster/variables.tf]: Minor change due to HCL2
```suggestion
  type        = number
```

File: modules/cluster/variables.tf [hunk 6]
```diff
@@ -29,12 +29,30 @@ variable "enabled_cluster_log_types" {
...
+  default     = 0
+}
+
+variable "manage_log_group" {
+  description = "Optionally manage the cluster log group via Terraform. Couple with `log_group_retention` to use a retention other than 'Never Expire'."
+  type        = "string"
```
Comment (johnctitus) [inline modules/cluster/variables.tf]: Minor change due to HCL2
```suggestion
  type        = bool
```

File: modules/cluster/examples/example.tf (hunk 1 above)
Comment (dannyleesmith) [reply modules/cluster/examples/example.tf]: In Terraform resources that's true but in Rackspace process standards when we use Rackspace modules we try to keep the vars in alphabetical order so it matches with the README.md

File: modules/cluster/main.tf (hunk 2 above)
Comment (dannyleesmith) [reply modules/cluster/main.tf]: I did this as common standards for resources would put `tags` last (as it's typically a map) after other inputs but before things like `depends_on`, `lifecycle`, etc.

File: modules/cluster/main.tf (hunk 3 above)
Comment (dannyleesmith) [reply modules/cluster/main.tf]: I did this as common standards for resources would put tags last (as it's typically a map) after other inputs but before things like depends_on, lifecycle, etc.

File: modules/cluster/variables.tf (hunk 4 above)
Comment (dannyleesmith) [reply modules/cluster/variables.tf]: Thanks, fixing

File: modules/cluster/variables.tf (hunk 5 above)
Comment (dannyleesmith) [reply modules/cluster/variables.tf]: Thanks, fixing

File: modules/cluster/variables.tf (hunk 6 above)
Comment (dannyleesmith) [reply modules/cluster/variables.tf]: Thanks, fixing

File: modules/cluster/examples/example.tf (hunk 1 above)
Comment (johnctitus) [reply modules/cluster/examples/example.tf]: Might be worth further discussion offline, but until that can occur and the defined style guidelines are updated, I would prefer we continue to treat all resources (module, explicit resource, and data resource) the same way within the modules.

Comment (dannyleesmith) [reply modules/cluster/examples/example.tf]: Agreed to resolve but actually did this by accident as `custom_tags` is now `tags` so is now at the end.
